sort_sorttype.go

```diff
@@ -190,13 +201,14 @@ func (s *SortTypeSorter) sortChunks() error {
 func (s *SortTypeSorter) saveChunks() error {
 	var err error
 	scratch := make([]byte, binary.MaxVarintLen64)
+	var raw []byte
 	for {
 		select {
 		case b, more := <-s.saveChunkChan:
 			if more {
 				for _, d := range b.data {
 					// binary encoding for size
-					raw := d.ToBytes()
+					raw = d.ToBytes(raw[:0])
```

Comment:
Why is this being done? it seems inefficient and confusing.
This should be 2 different variables to be more clear, although I'm still not convinced its a good way to do this.